app/src/main/java/com/google/android/sambadocumentsprovider/document/LoadDocumentTask.java

```diff
@@ -40,7 +40,11 @@ public LoadDocumentTask(Uri uri, SmbClient client, DocumentCache cache,
 
   @Override
   public DocumentMetadata run(Void... args) throws Exception {
-    return DocumentMetadata.fromUri(mUri, mClient);
+    if (mUri.getPathSegments().isEmpty()) {
```

Comment:
When queryChildDocuments is called for a file share we need to do metadata.createShare instead of fromUri because we can't get stat for a file share.